docs/proposals/operator-sdk-v1-migration.md

```diff
@@ -0,0 +1,285 @@
+
+### RBAC
+
+RBAC is maintained via kubebuilder RBAC Markers, rather than separate YAML files. The RBAC markers
+are placed on the `Reconcile` method of each controller, alongside implementing code. SDK projects
+can then generate RBAC for the controllers via a make target (`make manifests`). RBAC YAML has been
+moved to `config/rbac/` directory, rather than living alongside deployment manifests in `deploy`.
+
+New operators are cluster-scoped. We believe this is the right scoping for the build controllers.
+
+[6] https://sdk.operatorframework.io/docs/building-operators/golang/migration/#set-the-rbac-permissions
+[7] https://sdk.operatorframework.io/docs/building-operators/golang/operator-scope/
+
+### Migrate main.go
+
+The main invocation has changed its default generated code with respect to leader election.
+Operator SDK has not deprecated `leader.Become` (yet), but it has moved new code to use the leader
+election libraries from controller-runtime [8].
+
+If we maintain usage of `leader.Become`, there are changes to default behavior which we need to
+account for:
+
+1. The default metrics endpoint now binds to port `:8080` instead of `:8383`. To use the old port,
+   we need to pass the `--metrics-addr` argument to our binary.
+2. `OPERATOR_NAME` and `POD_NAME` env vars are no longer used for leader election. Instead, a
+   leader election ID is provided to the controller manager on startup. We may be ignoring these
+   environment variables already by hard-coding the LeaderElectionID field when the controller
+   manager is started up (`build-operator-lock`).
+
+Our operator allows the leader election namespace to be customized via the
+`BUILD_OPERATOR_LEADER_ELECTION_NAMESPACE` environment variable. This is ultimately wired through
+to controller-runtime's leader election logic - this should remain stable post migration.
+
+In addition to the old metrics port, our current controller also exposes port `:8686` to serve the
+custom resource metrics. Altering the metrics port could be considered a breaking change, as we
+would alter the related Kubernetes service. It is unclear if we can continue this practice.
+
+Changing the LeaderElectionID would be a breaking change - we would use different ConfigMaps for
+leader election and would not be able to support in-place upgrades with our deployment manifest.
+
+[8] https://sdk.operatorframework.io/docs/building-operators/golang/migration/#migrate-maingo
+
+### Migrate tests
+
+`operator-sdk test` is removed in v1, instead opting for the envtest library alongside ginkgo and
+gomega [9]. The test framework aligns with the upstream kubebuilder project, which now allows
+integration tests to run without an actual cluster [10].
+
+Our current integration test suite assumes that we have a running cluster with Tekton installed.
+Setup would need to be modified such that we install the Tekton CRDs the project utilizes [11].
```

Comment:
I´m not sure this is a must. We didn´t rely on operator-sdk test pkgs, therefore we dont need to do anything here.

In general, they recommend the usage of `envtest` pkg, but this implies a big refactoring on our side. I´m not sure it worth it. Not doing this changes  have none implications because we dont rely at all on the operator-sdk for integration. But I´m open for discussing this.

Reply:
I agree that this is scope that we do not have to take on. The integration suite can sit as is, and take on `envtest` as a later migration.